### PR TITLE
Add unit test for gpstart for better code coverage.

### PR DESCRIPTION
**Pull request by Chibin**
* Initial stab into creating unit tests for gpstart

### CODE REVIEW

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+
+class GpStart(GpTestCase):
+    def setUp(self):
+        # because gpconfig does not have a .py extension,
+        # we have to use imp to import it
+        # if we had a gpconfig.py, this is equivalent to:
+        #   import gpconfig
+        #   self.subject = gpconfig
+        gpstop_file = os.path.abspath(os.path.dirname(__file__) + "/../../../gpstart")
```
Comment (melanieplageman) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py]: typo: gpstop -> gpstart

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            patch('gpstart.dbconn.DbURL'),
+            patch('gpstart.dbconn.connect'),
+            patch('gpstart.GpArray.initFromCatalog', return_value=self.gparray),
+            patch('gpstart.catalog.getCollationSettings', return_value=(None, None, None)),
+            patch('gpstart.GpDbidFile'),
+            patch('gpstart.GpEraFile'),
+            patch('gpstart.userinput'),
+        ])
+
+        self.mock_os = self.get_mock_obj('exists')
```
Comment (melanieplageman) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py]: Larry sez: I vote for a more clear association between a mock and its patch.  get_mock_obj() makes this association less clear to me.  Putting the mock on the patch itself seems better to me.

Comment (melanieplageman) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py]: Also, consider naming "mock_os" -> "mock_os_path_exists" or something similar

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        gpstart = self.subject.GpStart.createProgram(options, args)
+        return_code = gpstart.run()
+
+        self.assertEqual(self.mock_userinput.ask_yesno.call_count, 0)
+        self.subject.logger.info.assert_any_call('Starting Master instance in admin mode')
+        self.subject.logger.info.assert_any_call('Master Started...')
+        self.assertEqual(return_code, 0)
+
+
+def os_exists_check(arg):
```
Comment (melanieplageman) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py]: Larry sez: I really like this--super clear, nicely mocking the file system behavior

**Comment by Chibin** [comment]
> Larry sez: I vote for a more clear association between a mock and its patch. get_mock_obj() makes this association less clear to me. Putting the mock on the patch itself seems better to me.

I can rename the get_mock_obj to something else. What I noticed is that if you give a return value=Mock() or some variable directed to a Mock(), when you a new object is created in the program, it creates a new Mock() for you. Directly using the patched mock mitigates that.

When you do a patch, it returns a Mock with it that you can play around with.
Would get_mock_from_apply_patch() as a function make more sense?

I can be wrong with the statement above, but that's just what I noticed.

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py (hunk 1 above)
Comment (Chibin) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py]: Oh, woops.. copy pasta. 
Fixed, thanks!

File: gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py (hunk 2 above)
Comment (Chibin) [inline gpMgmt/bin/gppylib/test/unit/test_unit_gpstart.py]: > Also, consider naming "mock_os" -> "mock_os_path_exists" or something similar
Changed to `mock_os_path_exists`

**Pull request merged**